app/models/fetch_extracts_by_subject.rb

```diff
@@ -0,0 +1,39 @@
+class FetchExtractsBySubject < FetcherBase
+  def extracts(query, extract_ids)
+    subject_extracts_query = query.except(:user_id)
+
+    subject_ids_from_specified_extracts = Extract.where(id: extract_ids).pluck(:subject_id)
+    exact_subject_ids = subject_ids_from_specified_extracts | Array.wrap(query[:subject_id])
+    exact_and_additional_subject_ids = exact_and_additional_subject_ids(exact_subject_ids)
```

Comment:
I think we can remove this variable here and modify the method the `get_*_subject_extracts`